### PR TITLE
services can use other type

### PR DESCRIPTION
**Pull request by agduncan94**
Service has the concept of OTHER files which are the files included in the Files array. I'm open to other possible names.

https://github.com/dockstore/dockstore/issues/2157

### CODE REVIEW

File: dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java [hunk 1]
```diff
@@ -50,7 +50,12 @@ public boolean isRelevantFileType(FileType type) {
             return super.isRelevantFileType(type) || type == FileType.DOCKERFILE || type == FileType.NEXTFLOW;
         }
     },
-    SERVICE("service", "generic placeholder for services", FileType.DOCKSTORE_SERVICE_YML, FileType.DOCKSTORE_SERVICE_TEST_JSON, true);
+    SERVICE("service", "generic placeholder for services", FileType.DOCKSTORE_SERVICE_YML, FileType.DOCKSTORE_SERVICE_TEST_JSON, true) {
+        @Override
+        public boolean isRelevantFileType(FileType type) {
+            return super.isRelevantFileType(type) || type == FileType.OTHER;
```
Comment (denis-yuen) [inline dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java]: Is FileType.OTHER used by anything other than services? 
Maybe it should be FileType.DOCKSTORE_SERVICE_OTHER or FileTYPE.DOCKSTORE_SERVICE_MISC if not.

Comment (agduncan94) [reply dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java]: Currently only services, though I could see it as a default value used by other entities.

Comment (denis-yuen) [reply dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java]: So I guess that's a question. What are the situations we would want assorted files to be intermingled with other types (tools, workflows) and when would we not want them to be?

Comment (agduncan94) [reply dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java]: Going with DOCKSTORE_SERVICE_OTHER

**Comment by agduncan94** [comment]
@coverbeck That is a good point, I would've assumed there would only be DOCKSTORE_YML

**Pull request merged**